AirCasting/BluetoothSession/BluetoothSessionRecordingController.swift

```diff
@@ -56,33 +57,39 @@ class MobileAirBeamSessionRecordingController: BluetoothSessionRecordingControll
         Resolver.resolve(AirBeamConfigurator.self, args: device)
             .configureMobileSession(location: locationTracker.location.value?.coordinate ?? .undefined,
                                     completion: completion)
+        
         // Info: we're not changing the sessions `status` property here to `.RECORDING` because it is currently
         // being done by the MeasurementStreamStorage class automagically.
         // This is something we might want to change at some point.
+        
+        guard !isRecording else { return } // To make sure we are not recording more than one session at once
+        
+        if !(activeSessionProvider.activeSession?.session.locationless ?? true) {
+            self.locationTracker.start()
+        }
+        recordMeasurements(for: activeSessionProvider.activeSession!)
     }
     
-    func stopRecordingSession(with uuid: SessionUUID) {
-        // Database change should be performed for active and disconnected sessions
-        performDatabaseChange(for: uuid)
+    func stopRecordingSession(with uuid: SessionUUID, databaseChange: (MobileSessionStorage) -> Void) {
+        // Database change should be performed for both active and disconnected sessions
+        databaseChange(storage)
         
         guard let activeSession = activeSessionProvider.activeSession, activeSession.session.uuid == uuid else { return }
```

Comment:
`isRecording` won't set to `false` if this `guard` fails. I think it might be a bug. Also - should we put an assert here to make sure noone calls stop without a start first?

Reply:
It's not a bug. `isRecording` should already be set to false cause it means that `stopRecordingSession` was called for disconnected session. (So maybe it is a bit misleading, but I would keep it like that for now cause I don't have a better idea how to do it.)